MapboxCoreNavigation/RouteController.swift

```diff
@@ -21,10 +22,43 @@ open class RouteController: NSObject {
         public static let shouldDisableBatteryMonitoring: Bool = true
     }
 
-    lazy var navigator: Navigator = {
+    var navigator: Navigator?
+
+    private var isConfigured: Bool = false
+    /**
+     Creates a cache for tiles of the given version and configures the navigator to use this cache.
+     */
+    func configureNavigator(withTilesVersion tilesVersion: String) throws -> Navigator {
+        // ~/Library/Caches/tld.app.bundle.id/.mapbox/2020_08_08-03_00_00/
+        guard var tilesURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            preconditionFailure("No Caches directory to create the tile directory inside")
+        }
+        if let bundleIdentifier = Bundle.main.bundleIdentifier ?? Bundle.mapboxCoreNavigation.bundleIdentifier {
+            tilesURL.appendPathComponent(bundleIdentifier, isDirectory: true)
+        }
+        tilesURL.appendPathComponent(".mapbox", isDirectory: true)
+        tilesURL.appendPathComponent(tilesVersion, isDirectory: true)
+        // Tiles with different versions shouldn't be mixed, it may cause inappropriate Navigator's behaviour
+        try FileManager.default.createDirectory(at: tilesURL, withIntermediateDirectories: true, attributes: nil)
+        return configureNavigator(withURL: tilesURL, tilesVersion: tilesVersion)
+    }
+
+    func configureNavigator(withURL tilesURL: URL, tilesVersion: String) -> Navigator {
+        let endpointConfig = TileEndpointConfiguration(directions: directions, tilesVersion: tilesVersion)
+        let tilesConfig = TilesConfig(tilesPath: tilesURL.path,
+                                      inMemoryTileCache: nil,
+                                      mapMatchingSpatialCache: nil,
+                                      threadsCount: nil,
+                                      endpointConfig: endpointConfig)
+
         let settingsProfile = SettingsProfile(application: ProfileApplication.kMobile, platform: ProfilePlatform.KIOS)
-        return Navigator(profile: settingsProfile, config: NavigatorConfig(), customConfig: "", tilesConfig: TilesConfig())
-    }()
+
+        let electronicHorizonOptions = ElectronicHorizonOptions(length: 2000, expansion: 1, branchLength: 50, includeGeometries: true)
```

Comment:
These length:expansion:branchLength: values come from what the Android implementation i doing now. We may want to change them later based on feedback from Nav Native.

Reply:
BTW these values don't match what we include by default on the Android side 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/c8ae2296fe577d51531e27b89b2091f2f4d77761/libnavigation-base/src/main/java/com/mapbox/navigation/base/options/EHorizonOptions.kt#L79-L82

For reference these are also defined in https://docs.google.com/document/d/1zVVp-u4ncsXLOzjLfQcaobU49x2Is8K_hsRDKwwqnMY/edit#heading=h.ba0pii7uoro7